packages/tupaia-web/src/components/Modal.tsx

```diff
@@ -0,0 +1,67 @@
+/*
+ * Tupaia
+ * Copyright (c) 2017 - 2023 Beyond Essential Systems Pty Ltd
+ */
+import React, { ReactNode } from 'react';
+import { Dialog, Paper as MuiPaper, useTheme, useMediaQuery } from '@material-ui/core';
+import MuiCloseIcon from '@material-ui/icons/Close';
+import styled from 'styled-components';
+import { Button } from '@tupaia/ui-components';
+
+interface ModalProps {
+  children?: ReactNode;
+  onClose: () => void;
+  isOpen: boolean;
+}
+
+const Wrapper = styled.div`
+  text-align: center;
+  overflow-x: hidden;
+  padding: 2em;
+`;
+
+const CloseIcon = styled(MuiCloseIcon)`
+  width: 1.2em;
+  height: 1.2em;
+`;
+
+const CloseButton = styled(Button)`
+  background-color: transparent;
+  min-width: initial;
+  padding: 0;
+  position: absolute;
+  top: 0.6em;
+  right: 0.6em;
+  &:hover,
+  &:focus {
+    background-color: transparent;
+    border: 1px solid ${({ theme }) => theme.palette.text.primary};
```

Comment:
Can we just use the material ui active styles? The white border feels a bit jarring to me

Reply:
I have updated this - I would advise a more obvious focus style, for accessibility purposes, just because the contrast on the default active styles is quite low, but we can add that later